include/NovelRT/DotNet/RuntimeService.h

```diff
@@ -8,41 +8,36 @@
 #endif
 
 namespace NovelRT::DotNet {
-  class RuntimeService {
+  class RuntimeService : public std::enable_shared_from_this<RuntimeService> {
   private:
+    struct Exports {
+      void(*Initialise)();
+      void(*Teardown)();
+      void(*FreeObject)(intptr_t obj);
+      void(*FreeString)(const char* str);
+      void(*GetInkServiceExports)(struct Ink::InkService::Exports* exports);
+    };
```

Comment:
Minor nit: Could we not use `std::add_pointer<T>::type` here to make these fnptrs easier to read?

Reply:
Possibly, but I think it would be better to just keep them like this as we will need these signatures anyways for use in the C wrapper (if we made it not a nested type, its definition could even be shared).